src/main.c

```diff
@@ -1565,14 +1579,21 @@ static void on_accept(h2o_socket_t *listener, const char *err)
 
     do {
         h2o_socket_t *sock;
-        if (num_connections(0) >= conf.max_connections) {
+        if (conf.soft_connection_limit > 0) {
```

Comment:
I think we might want to set `conf.soft_connection_limit` to `MAX_INT` to avoid this `if`.